refinery/data_set_manager/views.py

```diff
@@ -1059,11 +1059,26 @@ def post(self, request):
             return HttpResponseForbidden()
 
         logger.debug("Adding files to data set '%s'", data_set)
-        for file_store_item in data_set.get_file_store_items():
-            if (file_store_item.source.startswith(
+        node = get_object_or_404(Node, uuid=request.data.get('node_uuid'))
+
+        if node not in data_set.get_nodes():
```

Comment:
This API deals with Node instances and since each node is associated with exactly one data set then there should be a way to check permissions without requiring a `data_set_uuid` in the request.